ihmc-high-level-behaviors/src/main/java/us/ihmc/behaviors/activeMapping/ContinuousHikingStateMachine/WaitingToLandState.java

```diff
@@ -1,106 +1,66 @@
    private final ControllerFootstepQueueMonitor controllerQueueMonitor;
-
-   private final AtomicReference<FootstepStatusMessage> latestFootstepStatusMessage = new AtomicReference<>();
+   private FootstepStatusMessage previousFootstepStatusMessage = null;
+   private final ContinuousPlannerStatistics statistics;
 
    public WaitingToLandState(ROS2Helper ros2Helper,
                              String simpleRobotName,
-                             HumanoidReferenceFrames referenceFrames,
                              ContinuousPlanner continuousPlanner,
                              ControllerFootstepQueueMonitor controllerQueueMonitor,
-                             ContinuousHikingParameters continuousHikingParameters)
+                             ContinuousHikingParameters continuousHikingParameters,
+                             ContinuousPlannerStatistics statistics)
    {
       this.ros2Helper = ros2Helper;
-      this.referenceFrames = referenceFrames;
       this.continuousHikingParameters = continuousHikingParameters;
       this.continuousPlanner = continuousPlanner;
       this.controllerQueueMonitor = controllerQueueMonitor;
-
-      controllerQueueMonitor.attachFootstepStatusMessageConsumer(this::footstepStatusReceived);
+      this.statistics = statistics;
 
       controllerFootstepDataTopic = HumanoidControllerAPI.getTopic(FootstepDataListMessage.class, simpleRobotName);
       ros2Helper.createPublisher(controllerFootstepDataTopic);
    }
 
-   /*
-    * Callback to receive a message each time there is a change in the FootstepStatusMessage
-    */
-   private void footstepStatusReceived(FootstepStatusMessage footstepStatusMessage)
-   {
-      if (!continuousHikingParameters.getEnableContinuousWalking())
-         return;
-
-      if (footstepStatusMessage.getFootstepStatus() == FootstepStatusMessage.FOOTSTEP_STATUS_COMPLETED)
-      {
-         // TODO: Use the transfer time (starting now) to start planning (if WAITING_TO_LAND then plan again)
-         statistics.incrementTotalStepsCompleted();
-
-         double distance = referenceFrames.getSoleFrame(RobotSide.LEFT)
-                                          .getTransformToDesiredFrame(referenceFrames.getSoleFrame(RobotSide.RIGHT))
-                                          .getTranslation()
-                                          .norm();
-         statistics.setLastLengthCompleted((float) distance);
-
-         statistics.logToFile(true, true);
-      }
-
-      this.latestFootstepStatusMessage.set(footstepStatusMessage);
-   }
-
-   @Override
-   public boolean isDone(double timeInState)
-   {
-      FootstepStatusMessage footstepStatusMessage = latestFootstepStatusMessage.get();
-      return footstepStatusMessage.getFootstepStatus() == FootstepStatusMessage.FOOTSTEP_STATUS_STARTED;
-   }
-
    @Override
    public void onEntry()
    {
+      LogTools.warn(String.format("Entering %s", getClass().getSimpleName()));
```

Comment:
What is the purpose of this? If you want to print every state change, yoiu should set that up at the state machine level. Otherwise, you should remove this. This can be done by: 

factory.addStateChangedListener((from, to) -> LogTools.warn("Going from " + from.getClass().getSimpleName() + " to " + to.getClass().getSimpleName()));

Reply:
Ooo nice! Yep did that